### PR TITLE
[Fix] Only datetime and datetimev2 types can use current_timestamp as column default value

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
for this kind of sql:
```sql
create table test_default10(a int, b varchar(100) default current_timestamp) distributed by hash(a) properties('replication_num'="1");
```
add check:
 Types other than DATETIME and DATETIMEV2 cannot use current_timestamp as the default value

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run build all

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run p0

**Review by yiguolei — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -542,6 +542,16 @@ public static void validateDefaultValue(Type type, String defaultValue, DefaultV
             default:
                 throw new AnalysisException("Unsupported type: " + type);
         }
+        if (null != defaultValueExprDef && defaultValueExprDef.getExprName().equals("now")) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:545]: could we use this method directly? `isCurrentTimeStamp`

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run pipelines_p0

**Review by morrySnow — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**